### ISSUE TITLE
Why average the loss when training on multi-GPUs

### ISSUE THREAD

**Issue by pkuyym** [huggingface/pytorch-pretrained-BERT#411]
Could anyone help to explain the motivation for this operation?

https://github.com/huggingface/pytorch-pretrained-BERT/blob/master/examples/run_classifier.py#L573-L574

**Comment by thomwolf** [comment]
Multi-GPU loss returns a tuple of losses with one loss for each GPU.
We average them. See [this blog post](https://medium.com/huggingface/training-larger-batches-practical-tips-on-1-gpu-multi-gpu-distributed-setups-ec88c3e51255) for more details.
